src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Extensions/StartupExtensions.cs

```diff
@@ -17,12 +18,45 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Microsoft.OpenApi.Models;
 
 namespace DevAdventCalendarCompetition.Extensions
 {
     public static class StartupExtensions
     {
+        public static IServiceCollection AddAdventConfiguration(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            services.Configure<AdventSettings>(settings =>
+                {
+                    string defaultDateTimeFormat = "dd-MM-yyyy";
+                    var adventEndDate = configuration.GetSection("Competition:EndDate").Value;
+                    var adventStartDate = configuration.GetSection("Competition:StartDate").Value;
+                    var isValidStartDateTime = DateTimeOffset.TryParseExact(adventStartDate, defaultDateTimeFormat,
+                        CultureInfo.InvariantCulture,
+                        DateTimeStyles.None, out var startDate);
+                    var isValidEndDateTime = DateTimeOffset.TryParseExact(adventEndDate, defaultDateTimeFormat,
+                        CultureInfo.InvariantCulture,
+                        DateTimeStyles.None, out var endDate);
+
+                    if (!isValidStartDateTime || !isValidEndDateTime)
+                    {
+                        throw new InvalidOperationException(ExceptionsMessages.WrongFormatOfDate);
```

Comment:
you are not passing defaultDateTimeFormat parameter to this resource

Reply:
![image](https://user-images.githubusercontent.com/28907927/91844188-490b2b00-ec57-11ea-9f8a-866691cbdbc7.png)